mllib/src/main/scala/org/apache/spark/ml/tree/treeParams.scala

```diff
@@ -75,6 +75,23 @@ private[ml] trait DecisionTreeParams extends PredictorParams
     " discretizing continuous features.  Must be at least 2 and at least number of categories" +
     " for any categorical feature.", ParamValidators.gtEq(2))
 
+  /**
+   * If true, the trained tree will undergo a 'pruning' process after training in which nodes
+   * that have the same class predictions will be merged.  The benefit being that at prediction
+   * time the tree will be 'leaner'
```

Comment:
I think the key point for users is that this is a good thing _if_ only interested in class predictions. If interested in class probabilities, it won't necessarily give the right result and should be set to `false`. The text here is fine just wanting to make the tradeoffs explicit. I.e. `leaner` means smaller and faster.